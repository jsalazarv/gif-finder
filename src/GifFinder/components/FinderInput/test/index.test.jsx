import { fireEvent, render, screen } from '@testing-library/react';
import { FinderInput } from '../index';

describe('[FinderInput]', () => {
  describe('[Snapshots]', () => {
    test('It should contain [finder] class', () => {
      const { container } = render(<FinderInput />);

      expect(container.firstChild).toHaveClass('finder');
      expect(container).toMatchSnapshot();
    });
  });

  describe('[Events]', () => {
    test('Should change the value of the input', () => {
      render(<FinderInput onAddCategories={() => {}} />);
      const input = screen.getByRole('textbox');

      fireEvent.input(input, { target: { value: 'search test' } });

      expect(input.value).toBe('search test');
    });

    test('Should call [onAddCategories] if the form is submitted', () => {
      const inputValue = 'search test';
      const onAddCategories = jest.fn();

      render(<FinderInput onAddCategories={onAddCategories} />);

      const input = screen.getByRole('textbox');
      const form = screen.getByRole('form');

      fireEvent.input(input, { target: { value: 'search test' } });
      fireEvent.submit(form);

      expect(onAddCategories).toHaveBeenCalled();
      expect(onAddCategories).toHaveBeenCalledTimes(1);
      expect(onAddCategories).toHaveBeenCalledWith(inputValue);
    });

    test('Should not call [onAddCategories] if the input is empty', () => {
      const onAddCategories = jest.fn();

      render(<FinderInput onAddCategories={onAddCategories} />);

      const form = screen.getByRole('form');
      fireEvent.submit(form);

      expect(onAddCategories).toHaveBeenCalledTimes(0);
      expect(onAddCategories).not.toHaveBeenCalled();
    });
  });
});
