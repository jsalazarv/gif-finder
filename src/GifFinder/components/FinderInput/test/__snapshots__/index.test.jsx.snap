// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FinderInput] [Snapshots] It should contain [finder] class 1`] = `
<div>
  <div
    class="finder"
  >
    <form
      role="form"
    >
      <div
        class="wrapper"
      >
        <input
          placeholder="Hello, search all the GIFs..."
          type="text"
          value=""
        />
        <span
          class="underline"
        />
      </div>
    </form>
  </div>
</div>
`;
